frame/auction.ts

```diff
@@ -39,3 +62,48 @@ function randomToken() {
     )
     .join("");
 }
+
+async function fetchAndValidateTrustedScoringSignals(
+  baseUrl: string,
+  hostname: string,
+  renderingUrls: ReadonlySet<string>
+) {
+  const url = new URL(baseUrl);
+  url.searchParams.append("hostname", hostname);
+  url.searchParams.append(
+    "keys",
+    [...renderingUrls].map(encodeURIComponent).join(",")
+  );
+  const response = await tryFetchJson(url.href);
+  switch (response.status) {
+    case FetchJsonStatus.OK: {
+      const signals = response.value;
+      if (!isKeyValueObject(signals)) {
+        logError(`Expected JSON object but received ${jsonTypeOf(signals)}`);
```

Comment:
Would it be safe to log the whole thing?

Reply:
Yes; it can't leak through to the parent origin. I figured that it's better to keep log messages short and if developers want to know the actual value they can look in the network tab, but I could be convinced otherwise.